src/frontend/src/hooks/projects.hooks.ts

```diff
@@ -146,3 +147,19 @@ export const useAllLinkTypes = () => {
     return data;
   });
 };
+
+/**
+ * Custom React Hook to create a link type
+ */
+export const useCreateLinkType = () => {
+  const queryClient = useQueryClient();
+  return useMutation<LinkType, Error, { name: string }>(['linkTypes', 'create'], async (linkTypeData: { name: string }) => {
+    const { data } = await createLinkType(linkTypeData);
+    queryClient.invalidateQueries(['linkTypes']);
```

Comment:
do this how we do above with onSuccess. I have a hunch doing it this way causes issues, so I wanna avoid it